### ISSUE TITLE
Node security check falls

### ISSUE THREAD

**Issue by matrus2** [MikaAK/s3-plugin-webpack#72]
During the node security check found one vulnerability. 
```
(+) 1 vulnerabilities found
 Name     Installed   Patched   Path                                                                                                       More Info                             
 semver   2.2.1       >=4.3.2   webpack-s3-plugin@1.0.0-rc.0 > cdnizer@1.1.6 > jsdelivr-cdn-data@0.1.1 > semver@2.2.1   https://nodesecurity.io/advisories/31 
```

I've opened the PR at the end of this chain, hopefully it will be merged shortly. 
github/shahata/jsdelivr-cdn-data#7

Dependencies:
node v7.8.0
OS X 10.11.2

Step to reproduce:
- Install [nsp](https://github.com/nodesecurity/nsp)
- run `nsp check`

I see that I am not the first who is trying to solve this issue with `jsdelivr-cdn-data`. I would suggest to give it a try, though.

**Comment by d3viant0ne** [comment]
@matrus2 - This will be resolved in #78 one way or another.

**Comment by dcocchia** [comment]
This is also an issue with the `s3@4.4.0` dependency, as it depends on an outdated, insecure version of `mime`.

```
┌────────────┬────────────────────────────────────────────────────────────────────┐
│            │ Regular Expression Denial of Service                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Name       │ mime                                                               │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ CVSS       │ 7.5 (High)                                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Installed  │ 1.2.11                                                             │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Vulnerable │ < 1.4.1 || > 2.0.0 < 2.0.3                                         │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Patched    │ >= 1.4.1 < 2.0.0 || >= 2.0.3                                       │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ Path       │ instacart@1.0.0 > webpack-s3-plugin@1.0.0-rc.0 > s3@4.4.0 >        │
│            │ mime@1.2.11                                                        │
├────────────┼────────────────────────────────────────────────────────────────────┤
│ More Info  │ https://nodesecurity.io/advisories/535                             │
└────────────┴────────────────────────────────────────────────────────────────────┘
```

**Comment by d3viant0ne** [comment]
So here's the deal. We have been pinging the owners of those dependencies for months with no response. While i'm usually against doing this, we are going to have to fork or roll in the libs that aren't responding to move this forward.

**Comment by MikaAK** [comment]
For `s3` it's probably best to just replace it with `aws-sdk` and rework the code a bit to support it.

**Comment by d3viant0ne** [comment]
Agreed. Everything in contrib has a major release coming anyway to drop Node 4 support, we could roll the required changes into that.

I really wish people would take the security thing more seriously, I hate it when I get a CI / CD failure for something that could have been solved by simply maintaining deps.

I've got zero bandwidth for the s3 thing until i'm finished working on the closure plugin, next week at the absolute earliest.

In the mean time, i'll continue to try and reach out to the maintainers of the libs having nsp issues.
